static/module/list.js

```diff
@@ -120,28 +124,31 @@ export class List {
     // Only push state if URL is actually changing
     if (window.location.search !== queryString_) {
       const target = '/' + queryString_
-      const title = value.length > 0 ? `Search — ${value}` : this.initialTitle
+      const title = hasQuery ? `Search — ${query}` : this.initialTitle
```

Comment:
It already pushes to history but here it doesn't change the title.  Obviously, I had no query to show.  but not changing the title but pushing is not an option ...  and not pushing is not correct either ... so we need a title.